credentials/generate-revocation-set.py

```diff
@@ -64,6 +64,31 @@ def extract_single_integer_attribute(subject, oid):
     return None
 
 
+def extract_single_attribute_from_cn(cn, marker):
+    val_len = 4
+    start_idx = cn.find(marker)
+
+    if start_idx != -1:
+        val_start_idx = start_idx + len(marker)
+        return int(cn[val_start_idx:val_start_idx + val_len], 16)
+
+    return None
+
+
+def parse_vid_pid_from_distinguished_name(distinguished_name):
+    # VID/PID encoded using Matter specific RDNs
+    vid = extract_single_integer_attribute(distinguished_name, OID_VENDOR_ID)
+    pid = extract_single_integer_attribute(distinguished_name, OID_PRODUCT_ID)
+
+    # Fallback method to get the VID/PID, encoded in CN as "Mvid:FFFF Mpid:1234"
+    if vid is None and pid is None:
+        cn = distinguished_name.get_attributes_for_oid(x509.ObjectIdentifier("2.5.4.3"))[0].value
```

Comment:
Add `from cryptography.x509.oid import NameOID` at the top, and you can replace `"2.5.4.3"` with `NameOID.COMMON_NAME`